### PR TITLE
refactor!: YAML Config 

### PR DESCRIPTION
**Pull request by asanehisa**
WIP

### CODE REVIEW

**Comment by asanehisa** [comment]
Questions: 
Am I suppose to remove / reconfig sites-stream.json as well?
Have we decided if we are replacing yext.rc with config.yaml? 
Are the alpha code changes the Spruce stuff?

File: packages/pages/src/common/src/feature/artifacts.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+ * The shape of data that represents a artifacts.config file, used by Yext Pages.
```
Comment (mkilpatrick) [inline packages/pages/src/common/src/feature/artifacts.ts:2]: artifacts.json

File: packages/pages/src/common/src/feature/features.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { unsecureHashPluginName } from "../function/internal/types.js";
 import { defaultProjectStructureConfig } from "../project/structure.js";
 
 /**
- * The shape of data that represents a features.json file, used by Yext Pages.
+ * The shape of data that represents a templates.config (or features.json) file, used by Yext Pages.
```
Comment (mkilpatrick) [inline packages/pages/src/common/src/feature/features.ts:8]: .json

File: packages/pages/src/dev/dev.ts [hunk 3]
```diff
@@ -38,17 +38,28 @@ export const devCommand = (program: Command) => {
       false
     )
     .option("--noInit", "Disables automatic yext init with .yextrc file")
-    .option("--noGenFeatures", "Disable feature.json generation step")
+    .option(
+      "--noGenFeatures",
+      "Disable templates.config (or features.json) generation step"
```
Comment (mkilpatrick) [inline packages/pages/src/dev/dev.ts:43]: .json

File: packages/pages/src/dev/server/server.ts [hunk 4]
```diff
@@ -35,17 +36,23 @@ export const createServer = async (
...
     },
+    filenamesConfig: {
+      templatesConfig: yaml ? "templates.config" : undefined,
+      ciConfig: yaml ? undefined : "ci.json",
+    },
   });
 
-  // Read features.json and set the default locale to the first locale listed
-  // Default to en if features.json cannot be read or there is no locales entry
+  // Read template.config (or features.json) and set the default locale to the first locale listed
```
Comment (mkilpatrick) [inline packages/pages/src/dev/server/server.ts:45]: Same

File: packages/pages/src/generate/features/createFeaturesJson.ts [hunk 5]
```diff
@@ -17,12 +17,11 @@ export const getFeaturesConfig = async (
     features.push(featureConfig);
     module.config.stream && streams.push({ ...module.config.stream });
   }
-
   return { features, streams };
 };
 
 /**
- * Generates a features.json from the templates.
+ * Generates templates.config (or features.json) from the templates.
```
Comment (mkilpatrick) [inline packages/pages/src/generate/features/createFeaturesJson.ts:24]: json

File: packages/pages/src/generate/features/features.ts [hunk 6]
```diff
@@ -33,11 +38,12 @@ export const featureCommand = (program: Command) => {
         true,
         false
       );
+      const fileName = options.yaml ? "templates.config" : "features.json";
```
Comment (mkilpatrick) [inline packages/pages/src/generate/features/features.ts:41]: json

**Review by mkilpatrick — COMMENTED** [review]
I know the item says to use a flag but Spruce is almost done with their work at this point. I started a thread to see if we need to support both ways or not.

https://yext.slack.com/archives/C05GX6ZPU2J/p1691440221950349

**Pull request closed without merging**